src/Avalonia.Visuals/Animation/Animators/BaseBrushAnimator.cs

```diff
@@ -31,7 +32,7 @@ public class BaseBrushAnimator : Animator<IBrush>
         /// The type of the animator to instantiate.
         /// </typeparam>
         public static void RegisterBrushAnimator<TAnimator>(Func<Type, bool> condition)
```

Comment:
I was thinking to remove this RegisterBrushAnimator, because it does not makes much sense, when we can't render custom brushes anyway. But I left it with additional TryCreateCustomRegisteredAnimator method that uses reflection (as previous implementation). Normally animations will be created with TryCreateGradientAnimator or TryCreateSolidColorBrushAnimator that do not use reflection.

Reply:
Also TryCreateCustomRegisteredAnimator has higher priority. So if user actually wants to setup custom animator globally for existing brush types, he can do it.

Reply:
But personally I would remove it, even though it's a breaking change.

Reply:
I wonder what would be a better way of doing Brush type detection?

Reply:
You mean, completely new brush + its rendering? It will be useful for somebody, I agree. Some effects like acrylic might be done in that way.
But before we have it, RegisterBrushAnimator does not make much sense. Unless developer wants to override animator for specific type. But we already have nice API for that case https://github.com/AvaloniaUI/Avalonia/pull/6082 (it's not global though).

Reply:
I made that BrushAnimator class because XAML doesnt have a clean way of psasing the correct Brush type coz they're always wrapped in IBrush when they reach the animations subsystem.. But idk, if we can remove it in another PR that would be great

Reply:
I don't see any problem with BrushAnimator. It's still required. The difference is that it knows about supported brush animators instead of using reflection. Also tries to convert key value types.